drivers/audio/st_dmic_i2s.c

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2018 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "st_dmic_i2s.h"
+#include "OpenPDMFilter.h"
+
+#define LOG_LEVEL CONFIG_ST_DMIC_I2S_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(st_dmic_i2s);
+
+#define NUM_RX_BLOCKS			4
+#define PDM_BLOCK_MAX_SIZE_BYTES	512
+
+K_MEM_SLAB_DEFINE(rx_pdm_mem_slab, PDM_BLOCK_MAX_SIZE_BYTES, NUM_RX_BLOCKS, 1);
+
+/* ST_DMIC_I2S private data */
+static struct st_dmic_i2s_pdata {
```

Comment:
Might be worth setting this as a device data already, if we're to support multi-instance at one point

Reply:
makes sense